lib/apis/v3/kubecontrollersconfig.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	KindKubeControllersConfiguration     = "KubeControllersConfiguration"
+	KindKubeControllersConfigurationList = "KubeControllersConfigurationList"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeControllersConfiguration contains the configuration for Calico Kubernetes Controllers.
+type KubeControllersConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the KubeControllersConfiguration.
+	Spec KubeControllersConfigurationSpec `json:"spec,omitempty"`
+	// Status of the KubeControllersConfiguration
+	Status KubeControllersConfigurationStatus `json:"status,omitempty"`
+}
+
+// KubeControllersConfigurationSpec contains the values of the Kubernetes controllers configuration.
+type KubeControllersConfigurationSpec struct {
+
+	// LogSeverityScreen is the log severity above which logs are sent to the stdout. [Default: Info]
```

Comment:
Were you going to add the default value for each field? (FWIW we do that in felixconfig)

Reply:
Good idea